feat: check email should be lowercase

evanyang1 opened the pull request:
Fixes #1464

Frontend check to make sure new user email is lowercase.

  -
  -
  -

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
<!-- Note, if your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images -->
<!-- If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag -->

<details>
<summary>Visuals before changes are applied</summary>

<!-- ![image](Paste_Your_Image_Link_Here_After_Attaching_Files) -->

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
![image](https://github.com/hackforla/VRMS/assets/31290895/591b2a61-50ae-47bd-ad8e-3ea7c0733824)

</details>

Commit:
Implement React Hook Form attrs

Commit:
remove unused import

Commit:
destructure errors from formstate, clean line spaces

Commit:
errors onBlur and onChange

Commit:
update error status and use helper text

Commit:
clean up

Commit:
add conditions for location

Commit:
validate github profile to test location validation

Commit:
remove conditional attempt, replace double bang

Commit:
confuse people with a nested ternerary

evanyang1 committed:
feat: check email should be lowercase

trillium committed:
fix: Update ProjectForm.js

Force helperText to have content to prevent layout shift on  validation errors

trillium commented:
My appologies Evan, I didn't get back to this soon enough to give you more direction here. We're not looking for a validation error statement, we're looking for the information sent to the backend to be transformed to lowercase on the api request:

```js
// client/Auth.js 
if (isEmailValid) {
      const userData = await checkUser(email, LOG_IN);
// transform to:
      const userData = await checkUser(email.toLowerCase(), LOG_IN);

      if (userData) {
...
```

dependabot[bot] committed:
build(deps): bump word-wrap from 1.2.3 to 1.2.5 in /client

Bumps [word-wrap](https://github.com/jonschlinkert/word-wrap) from 1.2.3 to 1.2.5.
- [Release notes](https://github.com/jonschlinkert/word-wrap/releases)
- [Commits](https://github.com/jonschlinkert/word-wrap/compare/1.2.3...1.2.5)

---
updated-dependencies:
- dependency-name: word-wrap
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump word-wrap from 1.2.3 to 1.2.5 in /backend

Bumps [word-wrap](https://github.com/jonschlinkert/word-wrap) from 1.2.3 to 1.2.5.
- [Release notes](https://github.com/jonschlinkert/word-wrap/releases)
- [Commits](https://github.com/jonschlinkert/word-wrap/compare/1.2.3...1.2.5)

---
updated-dependencies:
- dependency-name: word-wrap
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

trillium reviewed (changes requested):
Talked over slack, but requesting changes to make transform the `email` auth api call to be `toLowerCase()`

evanyang1 committed:
!fix:
1. remove the previous commit's contents
2. convert to lowercase email in Auth.js

trillium commented:
@evanyang1 Can you please rebase this commit and push it back up again with only the necessary changes

```
# Rebase to current development
git fetch upstream
git rebase upstream/development


# rebase to edit whichn commits are used:
git rebase -i HEAD~2

# rebase contents should be:
# pick 9193340 feat: check email should be lowercase
# pick c58b97e !fix: 1. remove the previous commit's contents 2. convert to lowercase email in Auth.js

# change to
drop 9193340 feat: check email should be lowercase # drop this commit, old unused code
reword c58b97e !fix: 1. remove the previous commit's contents 2. convert to lowercase email in Auth.js 

# reword commmit c58b97e
feat: Add .toLowerCase for auth case senisitivity

```
